tests/test_redis_cache.py

```diff
@@ -247,3 +249,100 @@ def __str__(self):
         mock_client_object.assert_called_once_with(self.address, self.port)
         expected_hash = str(hash('cache_this_method' + str(test_class)))
         mock_client.get.assert_called_once_with(expected_hash)
+
+    def test_cache_on_class_without_str_for_function_with_simple_args(self):
+        """
+        Tests a cache hit when invoking a method that takes a primitive type as
+        argument on an object that does not implement a __str__ method
+        """
+        redis_cache = RedisCache(self.address, self.port)
+        redis_cache.redis_client = FakeRedisClient()
+
+        class TestClass(object):
+            def __init__(self):
+                self.call_count = collections.defaultdict(int)
+
+            @redis_cache.cache()
+            def echo(self, parameter):
+                self.call_count[parameter] += 1
+                return parameter
+
+        primitive_argument = 'cache hit test'
+
+        instance1 = TestClass()
```

Comment:
Maybe we should consider putting this in a loop instead of just calling the class/method 3 times. This is sort of a data driven test, so I think it should possibly be treated as such